### PR TITLE
Refactor/emotion

### PR DESCRIPTION
**Pull request by vuvu2259**

### CODE REVIEW

File: src/utils/rc-notification.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          {type === 'success' && <div className="title">Success</div>}
+          {type === 'error' && <div className="title">Error</div>}
+          <div className="desc">{content}</div>
+        </div>
+      </NotiContent>
+    ),
+    duration: 3,
+    prefixCls: 'cds',
+  });
+};
```
Comment (haVincy) [inline src/utils/rc-notification.tsx:37]: I think we can make type an `enum` as well.

```ts
export enum NotificationType {
   Success = 'Success',
   Error = 'Error'
}
```

File: example/index.tsx [hunk 2]
```diff
@@ -33,7 +31,7 @@ const App = () => {
       getTorusWallet({
         options: {
           clientId:
-            "BOM5Cl7PXgE9Ylq1Z1tqzhpydY0RVr8k90QQ85N7AKI5QGSrr9iDC-3rvmy0K_hF0JfpLMiXoDhta68JwcxS1LQ",
+            'BOM5Cl7PXgE9Ylq1Z1tqzhpydY0RVr8k90QQ85N7AKI5QGSrr9iDC-3rvmy0K_hF0JfpLMiXoDhta68JwcxS1LQ',
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply example/index.tsx:34]: ditto

File: src/api/backend/StatsAPI.ts [hunk 3]
```diff
@@ -1,23 +1,24 @@
-import axiosInstance from "../../config/axiosInstance";
+import axiosInstance from '../../config/axiosInstance';
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/api/backend/StatsAPI.ts]: It takes a lot of time to resolve path each file, I've just change the files which I refactor. Consider when I have time 😂

File: src/components/BuyModal/BuyModalConfirmed.tsx [hunk 4]
```diff
@@ -1,10 +1,10 @@
+import styled from '@emotion/styled';
 import { PublicKey } from '@solana/web3.js';
-import { Space, Col, Row } from 'antd';
 import React, { useMemo } from 'react';
 import IconTick from '../../assets/IconTick';
+import imgDefault from '../../assets/img-default.png';
 import { formatDate } from '../../utils/format';
 import { ExplorerLink } from '../ExplorerLink';
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/components/BuyModal/BuyModalConfirmed.tsx:7]: ditto

File: src/components/BuyModal/index.tsx [hunk 5]
```diff
@@ -55,9 +54,7 @@ export const BuyModal: React.FC<BuyModalProps> = ({
       setStep(2);
     } catch (error) {
       // Show error and redirect to step 0 again
-      errorNotification(
-        new Error('Transaction failed. Please try again later.')
-      );
+      notification('Transaction failed. Please try again later.', 'error');
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/components/BuyModal/index.tsx:57]: Nice, but I don't code it, I will ask @vanminh1701 (mint) in order to refactor it!

File: src/components/CancelModal/CancelModalDetail.tsx [hunk 6]
```diff
@@ -1,29 +1,27 @@
...
 import imgDefault from '../../assets/img-default.png';
+import { notification } from 'utils/rc-notification';
 
-import './style.less';
+export interface CancelModalDetailProps {
+  onCancel: any;
+  candyShop: CandyShop;
+  order: OrderSchema;
+  // eslint-disable-next-line no-unused-vars
+  onChangeStep: (...args: any) => void;
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/components/CancelModal/CancelModalDetail.tsx:17]: ditto

File: src/components/CancelModal/CancelModalDetail.tsx [hunk 7]
```diff
@@ -36,61 +34,148 @@ export const CancelModalDetail = ({
         onChangeStep(2);
       })
       .catch(() => {
-        errorNotification(
-          new Error('Transaction failed. Please try again later.')
-        );
+        notification('Transaction failed. Please try again later.', 'error');
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/components/CancelModal/CancelModalDetail.tsx:37]: ditto with @vanminh1701 (mint)

File: src/components/ExplorerLink/index.tsx [hunk 8]
```diff
@@ -32,13 +34,14 @@ export const ExplorerLink = (props: {
...
     >
-      {code ? (
+      {/* {code ? (
         <Typography.Text style={props.style} code>
           {shortenAddress(address, length)}
         </Typography.Text>
       ) : (
         shortenAddress(address, length)
-      )}
+      )} */}
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/components/ExplorerLink/index.tsx:43]: Nice ^^ I will remove it

File: src/components/ExplorerLink/index.tsx [hunk 9]
```diff
@@ -10,13 +9,16 @@ function shortenAddress(address: string, chars = 4): string {
 export const ExplorerLink = (props: {
   address: string | PublicKey;
   type: string;
-  code?: boolean;
+  // code?: boolean;
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/components/ExplorerLink/index.tsx:12]: ditto

File: src/components/Modal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+  width?: number;
+}
+
+const Modal: React.FC<ModalProps> = ({
+  children,
+  onCancel,
+  width = 1000,
+}: ModalProps) => {
+  // const modalRef = useRef<HTMLDivElement>();
+  // useClickOutside(modalRef, onCancel);
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/components/Modal/index.tsx:17]: ditto

File: src/config/axiosInstance.ts [hunk 11]
```diff
@@ -8,4 +8,23 @@ const axiosInstance = axios.create({
...
+        config.baseURL = 'https://candy-shop.liqnft.com/api/';
+      }
+      return config;
+    },
+    function (error) {
+      // Do something with request error
+      return Promise.reject(error);
+    }
+  );
+};
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/config/axiosInstance.ts:30]: Yup! Nice refactor ^^

File: src/core/Stat/index.tsx [hunk 12]
```diff
@@ -17,80 +16,131 @@ export const Stat = ({
...
-  const [width, setWidth] = useState<number>(window.innerWidth);
 
-  // handle window size change
-  function handleWindowSizeChange() {
-    setWidth(window.innerWidth);
-  }
+  const floorPrice = useMemo(() => {
+    if (!stat) return 0;
+    return (Number(stat.floorPrice) / LAMPORTS_PER_SOL).toFixed(2);
+  }, [stat]);
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/core/Stat/index.tsx:25]: We don't really need useMemo just cuz it make me easy to write code with computed variables. I will remove useMemo ^^

File: src/core/Stat/index.tsx [hunk 13]
```diff
@@ -17,80 +16,131 @@ export const Stat = ({
...
+  const totalListed = useMemo(() => {
+    if (!stat) return 0;
+    return stat.totalListed;
+  }, [stat]);
 
-  let isMobile: boolean = width <= 600;
+  const totalVolume = useMemo(() => {
+    if (!stat) return 0;
+    return (Number(stat.totalVolume) / LAMPORTS_PER_SOL).toFixed(2);
+  }, [stat]);
```
[the remark this answers is not in the document]
Comment (vuvu2259) [reply src/core/Stat/index.tsx:35]: ditto

File: src/utils/rc-notification.tsx (hunk 1 above)
Comment (vuvu2259) [reply src/utils/rc-notification.tsx:37]: Of course! Good job Wen! ^^

File: src/components/BuyModal/index.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (vanminh1701) [reply src/components/BuyModal/index.tsx:57]: noted with thanks @haVincy @vuvu2259

File: src/api/backend/StatsAPI.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (haVincy) [reply src/api/backend/StatsAPI.ts]: kk,

File: src/components/BuyModal/BuyModalDetail.tsx [hunk 14]
```diff
@@ -27,12 +32,7 @@ const BuyModalDetail: React.FC<BuyModalDetailProps> = ({
         <div className="buy-modal-control">
           <div>
             <div className="candy-label">PRICE</div>
-            <Statistic
-              suffix="SOL"
-              value={(order?.price as any) / LAMPORTS_PER_SOL}
-              precision={2}
-              valueStyle={{ fontSize: 16, fontWeight: 'bold' }}
-            />
+            <Price>{orderPrice} SOL</Price>
```
[the remark this answers is not in the document]
Comment (haVincy) [reply src/components/BuyModal/BuyModalDetail.tsx:35]: @RustySol could you help to confirm here, what do we expect user do see when fetching order failed or something wrong on blockchain that cause us can't get the result? I think the 0 price is not a good choice?

File: src/core/Stat/index.tsx (hunk 12 above)
[the remark this answers is not in the document]
Comment (haVincy) [reply src/core/Stat/index.tsx]: Thanks! Just keep in mind, we sometime will need to use it base on we really need it.

File: src/components/BuyModal/BuyModalDetail.tsx (hunk 14 above)
[the remark this answers is not in the document]
Comment (RustySol) [reply src/components/BuyModal/BuyModalDetail.tsx:35]: yes please display them as `N/A`

[the remark this answers is not in the document]
Comment (RustySol) [reply src/components/BuyModal/BuyModalDetail.tsx:35]: nvm I fixed

[the remark this answers is not in the document]
Comment (RustySol) [reply src/components/BuyModal/BuyModalDetail.tsx:35]: oh sorry I missed this
order price if 0 then we show it as 0, because its seller's choice

**Pull request merged**